app/src/ui/lib/list/list-row.tsx

```diff
@@ -221,7 +221,17 @@ export class ListRow extends React.Component<IListRowProps, {}> {
         onBlur={this.onBlur}
         onContextMenu={this.onContextMenu}
       >
-        {children}
+        {
+          // HACK: When we have an ariaLabel we need to make sure that the
+          // child elements are not exposed to the screen reader, otherwise
+          // VoiceOver will decide to read the children elements instead of the
+          // ariaLabel.
+          this.props.ariaLabel === undefined ? (
+            children
+          ) : (
+            <div aria-hidden={true}>{children}</div>
```

Comment:
Thoughts. Does it hurt to always have a `div` here. It would be easier to read as 
```
<div aria-hidden={this.props.ariaLabel === undefined}> {children}</div>

Reply:
It's definitely more convenient, my reasoning was to keep changes to a minimum for lists that won't make use of this, in order to avoid regressions like the one you found in the repository list 😳